### DISCUSSION TITLE
Documentation for development of custom components (exporter)

### DISCUSSION THREAD

**Discussion by gayatrigunnala** [open-telemetry/opentelemetry-collector-contrib#18738] [Q&A]
Hello,
I am attempting  to write a custom exporter for opentelemetry-collector-contrib.
Please provide me documentation links for the same which would include:
1. Implementation details - Interfaces to be implemented 
2. build & run steps - for local development
3. Integration with collector
4. packaging etc.,

Thank you

**Comment by jpkrohling** [comment]
I know that a receiver isn't an exporter, but this might serve as starting point to understand what it takes to create a component:

https://opentelemetry.io/docs/collector/trace-receiver/

@Chinwendu20, is there anything you want to add to the topic? :-)

**Comment by Chinwendu20** [comment]
Sure @jpkrohling , thank you.

@gayatrigunnala feel free to use this tool:
https://github.com/Chinwendu20/otel_components_generator
To generate skeleton code for your exporter

Please read this to understand how you can integrate your component with the collector contrib, it covers build and run steps as well as "packaging" as well:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md
